### PR TITLE
fix(curriculum): clarify underscore variable and introduce range

### LINKED ISSUE #55343: Clarify the use of single underscore as variable and introduce `range`
In [step 11 of the bisection method project](https://www.freecodecamp.org/learn/scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/step-11) instructions ask to create a `for` loop. The hint suggests to use `_` as the loop variable to iterate over `range(max_iterations)`.

- The `range` function should be introduced (currently this is done in [step 3 of the tower of Hanoi](https://www.freecodecamp.org/learn/scientific-computing-with-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/step-3))
-  as well as the use of a single underscore to indicate a variable that is not used in the code (mentioned in [step 18 of the password generator](https://www.freecodecamp.org/learn/scientific-computing-with-python/learn-regular-expressions-by-building-a-password-generator/step-18), which comes after this project).

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55343

### CODE REVIEW

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md [hunk 1]
```diff
@@ -9,7 +9,9 @@ dashedName: step-11
 
 Now you'll repeatedly narrow down the interval by finding the midpoint of the current interval and comparing the square of the midpoint with the target value.
 
-For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times.
+For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times. To iterate you can use range function, it returns a sequence of numbers, starting from 0 by default, and increments by 1 (by default), and stops before a specified number.
+
+You should use `_` as a loop variable. It is a conventional way to indicate intentionally unused variables when you are not utilizing the loop variable.
```
Comment (zairahira) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md:14]:
```suggestion
You should use `_` as a loop variable. The `_` acts as a placeholder and is useful when you need to use a variable but don't actually need its value.
```

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md [hunk 2]
```diff
@@ -9,7 +9,9 @@ dashedName: step-11
 
 Now you'll repeatedly narrow down the interval by finding the midpoint of the current interval and comparing the square of the midpoint with the target value.
 
-For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times.
+For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times. To iterate you can use range function, it returns a sequence of numbers, starting from 0 by default, and increments by 1 (by default), and stops before a specified number.
```
Comment (zairahira) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md:12]: From the instruction, it seems the campers would utilise the `range(start, stop)` variation.
The tests are only valid for `range(stop)`.

You can either modify the instruction to match the test or vice verca.

**Review by zairahira — CHANGES_REQUESTED** [review]

**Review by zairahira — APPROVED** [review]

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md [hunk 3]
```diff
@@ -9,7 +9,9 @@ dashedName: step-11
 
 Now you'll repeatedly narrow down the interval by finding the midpoint of the current interval and comparing the square of the midpoint with the target value.
 
-For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times.
+For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times. To iterate you can use range function, it returns a sequence of numbers starting from 0 and stops before a specified number.
+
+You should use `_` as a loop variable. The `_` acts as a placeholder and is useful when you need to use a variable but don't actually need its value.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md:14]:
```suggestion
For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times.

For your loop, use `range` function, which generates a sequence of numbers you can iterate over. The syntax is `range(start, stop, step)`, where `start` is the starting integer (inclusive), `stop` is the last integer (not inclusive), and `step` is the difference between a number and the previous one in the sequence.

Also, use `_` as a loop variable. The `_` acts as a placeholder and is useful when you need to use a variable but don't actually need its value.
```

**Review by Dario-DC — CHANGES_REQUESTED** [review]
I would expand a bit on `range`.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md (hunk 3 above)
Comment (gagan-bhullar-tech) [reply curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md]: @Dario-DC thanks for suggestions. Updated PR.

File: curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md [hunk 4]
```diff
@@ -9,9 +9,11 @@ dashedName: step-11
 
 Now you'll repeatedly narrow down the interval by finding the midpoint of the current interval and comparing the square of the midpoint with the target value.
 
-For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times. To iterate you can use range function, it returns a sequence of numbers starting from 0 and stops before a specified number.
+For that, inside the `else` block, create a `for` loop that runs up to `max_iterations` times.
 
-You should use `_` as a loop variable. The `_` acts as a placeholder and is useful when you need to use a variable but don't actually need its value.
+For your loop, use `range` function, which generates a sequence of numbers you can iterate over. The syntax is `range(start, stop, step)`, where `start` is the starting integer (inclusive), `stop` is the last integer (not inclusive), and `step` is the difference between a number and the previous one in the sequence.
```
Comment (Dario-DC) [inline curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef1be2b958cabf712296cd.md:14]:
```suggestion
For your loop, use the `range` function, which generates a sequence of numbers you can iterate over. The syntax is `range(start, stop, step)`, where `start` is the starting integer (inclusive), `stop` is the last integer (not inclusive), and `step` is the difference between a number and the previous one in the sequence.
```
Oops, I missed the article in the suggestion

**Review by Dario-DC — APPROVED** [review]
LGTM :+1:

**Pull request merged**